webpack/assets/javascripts/react_app/components/HostsIndex/BulkActions/reassignHostGroup/BulkReassignHostgroupModal.js

```diff
@@ -0,0 +1,194 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { FormattedMessage } from 'react-intl';
+import {
+  Modal,
+  Button,
+  TextContent,
+  Text,
+  FormGroup,
+  FormSelect,
+  FormSelectOption,
+} from '@patternfly/react-core';
+import { addToast } from '../../../ToastsList/slice';
+import { translate as __ } from '../../../../common/I18n';
+import { failedHostsToastParams } from '../helpers';
+import { STATUS } from '../../../../constants';
+import {
+  selectAPIStatus,
+  selectAPIResponse,
+} from '../../../../redux/API/APISelectors';
+import {
+  BULK_REASSIGN_HOSTGROUP_KEY,
+  bulkReassignHostgroups,
+  fetchHostgroups,
+  HOSTGROUP_KEY,
+} from './actions';
+import { visit } from '../../../../../foreman_navigation';
+import { foremanUrl } from '../../../../common/helpers';
+
+const BulkReassignHostgroupModal = ({
+  isOpen,
+  closeModal,
+  selectedCount,
+  orgId,
+  fetchBulkParams,
+}) => {
+  const dispatch = useDispatch();
+  const [hostgroupId, setHostgroupId] = useState('');
+  const hostgroups = useSelector(state =>
+    selectAPIResponse(state, HOSTGROUP_KEY)
+  );
+  const hostgroupStatus = useSelector(state =>
+    selectAPIStatus(state, HOSTGROUP_KEY)
+  );
+  const hostUpdateStatus = useSelector(state =>
+    selectAPIStatus(state, BULK_REASSIGN_HOSTGROUP_KEY)
+  );
+  const handleModalClose = () => {
+    setHostgroupId('');
+    closeModal();
+  };
+
+  useEffect(() => {
+    dispatch(fetchHostgroups());
+  }, [dispatch]);
+
+  const handleError = ({ response }) => {
+    handleModalClose();
+    dispatch(
+      addToast(
+        failedHostsToastParams({
+          ...response.data.error,
+          key: BULK_REASSIGN_HOSTGROUP_KEY,
+        })
+      )
+    );
+  };
+  const handleSave = () => {
+    const requestBody = {
+      included: {
+        search: fetchBulkParams(),
+      },
+      hostgroup_id: hostgroupId,
+    };
+
+    dispatch(
+      bulkReassignHostgroups(
+        requestBody,
+        () => visit(foremanUrl('/new/hosts')),
```

Comment:
Not sure if there's an easy way to fix this, but I see the toast flash on screen only for a fraction of a second. And then it's obliterated by the page refresh.